packages/backend/src/services/warehouseClients/PostgresWarehouseClient.ts

```diff
@@ -44,4 +123,66 @@ export default class PostgresWarehouseClient implements QueryRunner {
     async test(): Promise<void> {
         await this.runQuery('SELECT 1');
     }
+
+    async getSchema(
+        requests: {
+            database: string;
+            schema: string;
+            table: string;
+            columns: string[];
+        }[],
+    ) {
+        const { databases, schemas, tables } = requests.reduce<{
+            databases: Set<string>;
+            schemas: Set<string>;
+            tables: Set<string>;
+        }>(
+            (acc, { database, schema, table }) => ({
+                databases: acc.databases.add(`'${database}'`),
+                schemas: acc.schemas.add(`'${schema}'`),
+                tables: acc.tables.add(`'${table}'`),
+            }),
+            {
+                databases: new Set(),
+                schemas: new Set(),
+                tables: new Set(),
+            },
+        );
+        const query = `
+            SELECT table_catalog,
+                   table_schema,
+                   table_name,
+                   column_name,
+                   data_type
+            FROM information_schema.columns
+            WHERE table_catalog IN (${Array.from(databases)}) 
+            AND table_schema IN (${Array.from(schemas)})
+            AND table_name IN (${Array.from(tables)})
+        `;
+
+        const rows = await this.runQuery(query);
+
+        return rows.reduce(
```

Comment:
Since the `rows` are reduced and not the original `requests` this will return more results that expected.

Reply:
Related to this answer: https://github.com/lightdash/lightdash/pull/716#discussion_r740319487